presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -128,6 +136,79 @@ public TopNNode map(TopNNode node, PlanNode source, PlanNodeId newNodeId)
                 node.getStep());
     }
 
+    public TableWriterNode map(TableWriterNode node, PlanNode source)
+    {
+        return map(node, source, node.getId());
+    }
+
+    public TableWriterNode map(TableWriterNode node, PlanNode source, PlanNodeId newNodeId)
```

Comment:
I tried to follow the planning bits but I don't understand this at all. Ask @martint, @kokosing or @sopel39 to review this commit.